pkg/app/api/httpapi/metricsmiddleware/instrument_handler.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricsmiddleware
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	codeLabel   = "code"
+	methodLabel = "method"
+	pathLabel   = "path"
+)
+
+var (
+	requestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_requests_total",
+			Help: "Total number of HTTP requests.",
+		},
+		[]string{codeLabel, methodLabel, pathLabel},
+	)
+
+	durationHistgram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "http_request_duration_milliseconds",
+			Help:    "Histogram of request latencies in milliseconds.",
+			Buckets: []float64{0.0001, 0.025, 0.1, 1, 25, 100},
+		},
+		[]string{codeLabel, methodLabel, pathLabel},
+	)
+)
+
+func Register(r prometheus.Registerer) {
+	r.MustRegister(
+		requestCounter,
+		durationHistgram,
+	)
+}
+
+func InstrumentHandler(path string, next http.Handler) http.Handler {
```

Comment:
nit: `Handler` is fine because the package name already says mean that this is for metrics.